cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,673 @@
+    - 'timedelta': the time difference with respect the reference time 0
+    - 'variable_name': the name of the variable (without the lag information).
+    """
+
+    def __init__(
+        self,
+        input_list: Optional[List[NodeLike]] = None,
+        output_list: Optional[List[NodeLike]] = None,
+        fully_connected: bool = True,
+        store_graphs: bool = False,
+    ):
+        """Initialize the time series causal graph."""
+        super().__init__(input_list, output_list, fully_connected)
+
+        self._store_graphs = store_graphs
+        # autoregressive order of the graph (max lag)
+        self._order: Optional[int] = None
+        # list of variables in the graph, i.e. discarding the lags (X1(t-1) and X1 are the same variable)
+        self._variables: Optional[List[str]] = None
+        self._summary_graph: Optional[CausalGraph] = None
+        self._minimal_graph: Optional[TimeSeriesCausalGraph] = None
+        self._meta_time_name = 'time_lag'
+        self._meta_variable_name = 'variable_name'
+
+    def __eq__(self, other: object) -> bool:
+        """Return True if the graphs are equal."""
+        if not isinstance(other, TimeSeriesCausalGraph):
+            return False
+
+        # now check if the graphs are equal. Since TimeSeriesCausalGraph is a subclass of CausalGraph,
+        # we can use the CausalGraph.__eq__ method and then check for the metadata
+        are_equal = super().__eq__(other)
+        # no need to check the metadata if the graphs are not equal
+        if not are_equal:
+            return False
+
+        # now check the metadata timedelta in the nodes
+        for node in self.get_nodes():
+            other_node = other.get_node(node.identifier)
+            # check the variable name
+            node_metadata_variable = node.get_metadata()[self._meta_variable_name]
+            other_node_metadata_td = other_node.get_metadata()[self._meta_variable_name]
+
+            if node_metadata_variable != other_node_metadata_td:
+                return False
+
+            # check the time delta
+            node_metadata_td = node.get_metadata()[self._meta_time_name]
+            other_node_metadata_td = other_node.get_metadata()[self._meta_time_name]
+
+            if node_metadata_td != other_node_metadata_td:
+                return False
+
+        return True
+
+    def copy(self, include_meta: bool = True) -> TimeSeriesCausalGraph:
+        """Return a copy of the graph."""
+        graph = super().copy(include_meta=include_meta)
+        # cast the graph to TimeSeriesCausalGraph to have the correct metadata
+        return TimeSeriesCausalGraph.from_causal_graph(graph)
+
+    def get_minimal_graph(self) -> TimeSeriesCausalGraph:
+        """
+        Return a minimal graph.
+
+        The minimal graph is the graph with the minimal number of edges that is equivalent to the original graph.
+        In other words, it is a graph that has no edges whose destination is not time delta 0.
```

Comment:
should include an example here I would say to make clear what the minimal graph is.